lib/msal-react/src/hooks/useMsalAuthentication.ts

```diff
@@ -48,10 +74,60 @@ export function useMsalAuthentication(
                 logger.verbose("useMsalAuthentication - Calling ssoSilent");
                 return instance.ssoSilent(loginRequest as SsoSilentRequest);
             default:
-                throw "Invalid interaction type provided.";
+                throw ReactAuthError.createInvalidInteractionTypeError();
         }
     }, [instance, interactionType, authenticationRequest, logger]);
 
+    const acquireToken = useCallback(async (callbackInteractionType?: InteractionType, callbackRequest?: SilentRequest): Promise<AuthenticationResult|null> => {
+        const fallbackInteractionType = callbackInteractionType || interactionType;
+
+        let tokenRequest: SilentRequest;
+
+        if (callbackRequest) {
+            tokenRequest = {
+                ...callbackRequest
+            };
+        } else if (authenticationRequest) {
+            tokenRequest = {
+                ...authenticationRequest,
+                scopes: authenticationRequest.scopes || OIDC_DEFAULT_SCOPES
+            };
+        } else {
+            tokenRequest = {
+                scopes: OIDC_DEFAULT_SCOPES
+            };
+        }
+        
+        if (!tokenRequest.account && account) {
+            tokenRequest.account = account;
+        }
+
+        const getToken = async (): Promise<AuthenticationResult|null> => {
+            logger.verbose("useMsalAuthentication - Calling acquireTokenSilent");
+            return instance.acquireTokenSilent(tokenRequest).catch(async (e: AuthError) => {
+                if (e instanceof InteractionRequiredAuthError) {
+                    if (!interactionInProgress.current) {
+                        logger.verbose("useMsalAuthentication - Interaction required, falling back to interaction");
+                        return login(fallbackInteractionType, tokenRequest);
+                    } else {
+                        logger.verbose("useMsalAuthentication - Interaction required but is already in progress. Please try again, if needed, after interaction completes.");
```

Comment:
I think these should be `logger.error`, given they are during an error case.

Reply:
Done